scene/gui/progress_bar.cpp

```diff
@@ -84,17 +99,40 @@ bool ProgressBar::is_percent_visible() const {
 	return percent_visible;
 }
 
+void ProgressBar::set_fill_mode(FillMode p_fill) {
+
+	fill = p_fill;
+	update();
+}
+
+ProgressBar::FillMode ProgressBar::get_fill_mode() const {
+
+	return fill;
+}
+
 void ProgressBar::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("set_percent_visible", "visible"), &ProgressBar::set_percent_visible);
 	ClassDB::bind_method(D_METHOD("is_percent_visible"), &ProgressBar::is_percent_visible);
+
+	ClassDB::bind_method(D_METHOD("get_fill_mode"), &ProgressBar::get_fill_mode);
+	ClassDB::bind_method(D_METHOD("set_fill_mode", "fill_direction"), &ProgressBar::set_fill_mode);
+
 	ADD_GROUP("Percent", "percent_");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "percent_visible"), "set_percent_visible", "is_percent_visible");
+
+	BIND_ENUM_CONSTANT(FILL_UP);
+	BIND_ENUM_CONSTANT(FILL_DOWN);
+	BIND_ENUM_CONSTANT(FILL_LEFT);
+	BIND_ENUM_CONSTANT(FILL_RIGHT);
+
+	ADD_GROUP("Fill Direction", "fill_");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "fill_direction", PROPERTY_HINT_ENUM, "Up,Down,Left,Right"), "set_fill_mode", "get_fill_mode");
 }
 
 ProgressBar::ProgressBar() {
-
 	set_v_size_flags(0);
 	set_step(0.01);
 	percent_visible = true;
-}
+	fill = FILL_RIGHT;
```

Comment:
Isn't it weird to have it named FILL_RIGHT if the bar is filled on its left ?

I would rename the enum to FillDirection and name the enum values like `FILL_DIR_<direction>`

Reply:
@groud  Yeah, I was actually looking at how it was done in TextureProgress and it was done like FILL_FROM_<start>_to_<end>, or something like that. I was thinking I should probably change it to that style for consistency.  Another thing I will probably change is making the "left to right" ENUM be 0, instead of 3. Seems more appropriate I think cause its the default value. I will probably do that all sometime this weekend though when I have some more time.

Reply:
Ok, it sounds good to me :)

Reply:
Okay, I updated the ENUM name. I think it will be a bit clearer this way. And it is consistent with the naming in TextureProgress.